File: 1413-minimum-value-to-get-positive-step-by-step-sum/1413-minimum-value-to-get-positive-step-by-step-sum.py
class Solution:
    def minStartValue(self, nums: List[int]) -> int:
        m = 1
        s = 0
        for i in nums:
            s += i
            if s<m and s!=0:
                m = s
        if m<0:
            return (m*-1)+1
        return m

File: 1413-minimum-value-to-get-positive-step-by-step-sum/1413-minimum-value-to-get-positive-step-by-step-sum.cpp
class Solution {
public:
    int minStartValue(vector<int>& nums) {
        int min = 1;
        int sum = 0;
        for(int i=0; i<nums.size(); i++){
            sum += nums[i];
            if(sum<min && sum!=0){
                min = sum;
            }
        }
        if(min<0){
            return -min+1;
        }
        else{
            return min;
        }
    }
};